libbeat/publisher/pipeline/pipeline_test.go

```diff
@@ -18,12 +18,104 @@
 package pipeline
 
 import (
+	"runtime"
 	"sync"
+	"testing"
 
+	"github.com/elastic/beats/v7/libbeat/beat"
 	"github.com/elastic/beats/v7/libbeat/common/atomic"
 	"github.com/elastic/beats/v7/libbeat/publisher/queue"
+	"github.com/elastic/beats/v7/libbeat/tests/resources"
+	"github.com/elastic/elastic-agent-libs/mapstr"
 )
 
+func TestPipelineAcceptsAnyNumberOfClients(t *testing.T) {
+	routinesChecker := resources.NewGoroutinesChecker()
+	defer routinesChecker.Check(t)
+
+	pipeline := makePipeline(t, Settings{}, makeDiscardQueue())
+
+	defer pipeline.Close()
+
+	n := 66000
+	clients := []beat.Client{}
+	for i := 0; i < n; i++ {
+		c, err := pipeline.ConnectWith(beat.ClientConfig{})
+		if err != nil {
+			t.Fatalf("Could not connect to pipeline: %s", err)
+		}
+		clients = append(clients, c)
+	}
+
+	for i, c := range clients {
+		c.Publish(beat.Event{
+			Fields: mapstr.M{
+				"count": i,
+			},
+		})
+	}
+
+	// Close the first 105 clients
+	nn := 105
+	clientsToClose := clients[:n]
+	clients = clients[nn:]
+
+	for _, c := range clientsToClose {
+		c.Close()
+	}
+
+	// Let other goroutines run
+	runtime.Gosched()
+	runtime.Gosched()
+
+	// Make sure all clients are closed
+	for _, c := range clients {
+		c.Close()
+	}
+}
+
+// makeDiscardQueue returns a queue that always discards all events
+// the producers are assigned an unique incremental ID, when their
+// close method is called, this ID is returned
+func makeDiscardQueue() queue.Queue {
+	var wg sync.WaitGroup
+	producerID := atomic.NewInt(0)
+
+	return &testQueue{
+		close: func() error {
+			//  Wait for all producers to finish
+			wg.Wait()
+			return nil
+		},
+		get: func(count int) (queue.Batch, error) {
+			return nil, nil
+		},
+
+		producer: func(cfg queue.ProducerConfig) queue.Producer {
+			producerID.Inc()
+			id := producerID.Load()
+
+			// count is a counter that increments on every published event
+			// it's also the returned Event ID
+			count := uint64(0)
+			producer := &testProducer{
+				publish: func(try bool, event interface{}) (queue.EntryID, bool) {
```

Comment:
this doesn't compile it needs to be 
`publish func(try bool, event queue.Entry) (queue.EntryID, bool)`